services/slam/server_test.go

```diff
@@ -149,19 +305,71 @@ func TestServer(t *testing.T) {
 		test.That(t, resp, test.ShouldBeNil)
 	})
 
+	t.Run("failing get pointcloud map stream function", func(t *testing.T) {
+		// GetPointCloudMapStreamFunc failure
+		injectSvc.GetPointCloudMapStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+			f := func() ([]byte, error) {
+				return []byte{}, io.EOF
+			}
+			return f, errors.New("failure to get pointcloud map")
+		}
+
+		reqPointCloudMapStream := &pb.GetPointCloudMapStreamRequest{Name: testSlamServiceName}
+
+		mockServer := makePointCloudStreamServerMock()
+		err = slamServer.GetPointCloudMapStream(reqPointCloudMapStream, mockServer)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "failure to get pointcloud map")
+
+		// Callback failure
+		injectSvc.GetPointCloudMapStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+			f := func() ([]byte, error) {
+				return []byte{}, errors.New("callback error")
+			}
+			return f, nil
+		}
+
+		mockServer = makePointCloudStreamServerMock()
+		err = slamServer.GetPointCloudMapStream(reqPointCloudMapStream, mockServer)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "callback error")
+	})
+
 	t.Run("failing get internal state function", func(t *testing.T) {
 		injectSvc.GetInternalStateFunc = func(ctx context.Context, name string) ([]byte, error) {
 			return nil, errors.New("failure to get internal state")
 		}
 
-		req := &pb.GetInternalStateRequest{
-			Name: testSlamServiceName,
-		}
+		req := &pb.GetInternalStateRequest{Name: testSlamServiceName}
 		resp, err := slamServer.GetInternalState(context.Background(), req)
 		test.That(t, err.Error(), test.ShouldContainSubstring, "failure to get internal state")
 		test.That(t, resp, test.ShouldBeNil)
 	})
 
+	t.Run("failing get internal state stream function", func(t *testing.T) {
+		// GetInternalStateStreamFunc error
+		injectSvc.GetInternalStateStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+			f := func() ([]byte, error) {
+				return []byte{}, io.EOF
+			}
+			return f, errors.New("failure to get internal state")
```

Comment:
I recommend making f nil in this case given that the callback function caller should never invoke the callback function given an error will always be returned.

Reply:
jk u right, changed these for client tests but not server tests